pkgs/development/libraries/atk/default.nix

```diff
@@ -17,9 +17,12 @@ stdenv.mkDerivation rec {
 
   outputs = [ "out" "dev" ];
 
-  buildInputs = stdenv.lib.optional stdenv.isDarwin fixDarwinDylibNames;
+  buildInputs = [
+    gobject-introspection
+    glib
```

Comment:
That does not look right to me. `gobject-introspection` is used during build for generating the GIR and `glib` is probably used for `glib-mkenums` or something else during the build (it was specifically added in ebdd6448369f015e3050f2253239a29f1f8a6f62).

Reply:
These not being available for target triple were causing the configure stage to fail, as such:

```
atk> WARNING: Could not detect glib version, assuming 2.54. You may get build errors if your glib is older.
atk> atk/meson.build:138:2: ERROR: dependency gobject-introspection-1.0 not found:
atk> Package gobject-introspection-1.0 was not found in the pkg-config search path.
atk> Perhaps you should add the directory containing `gobject-introspection-1.0.pc'
atk> to the PKG_CONFIG_PATH environment variable
atk> No package 'gobject-introspection-1.0' found
```

Meson logs reveal that it is looking at targets’ pkg-config for gobject-introspection.

It _might_ be the case that these two dependencies for both target and host are necessary, that the package being built has a buggy build definition etc, but I am unable to really verify that complete _cross-_compilation works with this change because there are a number of dependencies that fail to build before it.

That said, `glib` is already present in `propagatedBuildInputs` so moving `glib` from native to plain `buildInputs` is indeed superfluous.

I guess its better to close this for now and revisit once gobject-introspection works at all.

Reply:
Looks like it may be indeed a buggy meson file. I hacked around it a little to get configure to pass and during building it starts invoking executables from the target's gobject-introspection. I’ll take my bug there.

Reply:
The "Could not detect glib version" message comes from Meson itself, most likely from gnome.genmarshal function.

Reply:
https://github.com/mesonbuild/meson/blob/14a0f18d2a177d53e8175741bdb9de093241e4a4/mesonbuild/modules/gnome.py#L68-L75 looks reasonable. Maybe something to do with the recent pkg-config cross changes?

Reply:
Add a `depsBuildBuild = [ pkg-config ];` (or `pkgsBuildBuild.pkg-config` without splicing) to be able to resolve native pkg-config deps.